### ISSUE TITLE
the issue of reproduced performance 

### ISSUE THREAD

**Issue by khk-abc** [microsoft/TOXIGEN#29]
following the use:
"
from transformers import pipeline
toxigen_hatebert = pipeline("text-classification", model="tomh/toxigen_hatebert", tokenizer="bert-base-cased")
toxigen_hatebert("I love science")
"
or 
"
from transformers import AutoModelForSequenceClassification
toxigen_hatebert = AutoModelForSequenceClassification.from_pretrained("tomh/toxigen_hatebert")
"
the f1 is very small and the method cannot achieve the performance in the paper. using toxigen-val dataset.

by the way, the index 1 of logits means the positive label (i.e., toxic)?

my python scripts:
from transformers import AutoTokenizer, AutoModel
import json
from sklearn.metrics import f1_score,precision_score,recall_score,accuracy_score,roc_auc_score
from tqdm import tqdm
import torch
from transformers import pipeline

from transformers import AutoModelForSequenceClassification

with open(data_path,'r') as f:
    datas= json.load(f)

toxigen_hatebert = AutoModelForSequenceClassification.from_pretrained("pretrained_models/toxigen_hatebert")
tokenizer = AutoTokenizer.from_pretrained('pretrained_models/bert-base-cased')

\# toxigen_hatebert = pipeline("text-classification",
\#                             model="pretrained_models/toxigen_hatebert",
\#                             tokenizer="pretrained_models/bert-base-cased")

predicts = []
trues = []
toxigen_hatebert.eval()
with torch.no_grad():
    for idx,item in enumerate(tqdm(datas)):
        content = item['content']
        \# outputs = toxigen_hatebert(content)
        inputs = tokenizer(content,return_tensors='pt',padding='longest')
        outputs = toxigen_hatebert(**inputs)
        print(outputs)
        trues.append(item['toxic'])
        predicts.append(1-torch.argmax(outputs['logits'][0]).tolist())
        print('true: ', item['toxic'])
        print('pred: ', predicts[-1])

f1=f1_score(y_true=trues,y_pred=predicts)
precision=precision_score(y_true=trues,y_pred=predicts)
recall=recall_score(y_true=trues,y_pred=predicts)
acc=accuracy_score(y_true=trues,y_pred=predicts)
auc=roc_auc_score(y_true=trues,y_score=predicts)

print(f1,precision,recall,acc,auc)

**Comment by Thartvigsen** [comment]
Hi @khk-abc thanks for your interest in our work! Could you provide more details about what dataset you are using? I don't know what `item["content"]` and `item["toxic"]` contain. If `item["toxic"]` contains the binary toxicity label, how are you computing this? I believe that yes, `outputs.logits[0]` contains the logits for the non-toxic class. Can you also try using the "bert-based-uncased" tokenizer? I also suggest pasting code using code blocks to help others understand your code!

**Comment by khk-abc** [comment]
Thanks for your answering, item['content'] means the comment in the toxigen-val dataset (total 940 samples)，item['toxic'] means the ground truth, 1 denotes the comment is toxic













At 2024-01-17 16:16:48, "Tom Hartvigsen" ***@***.***> wrote:

Hi @khk-abc thanks for your interest in our work! Could you provide more details about what dataset you are using? I don't know what item["content"] and item["toxic"] contain. If item["toxic"] contains the binary toxicity label, how are you computing this? I believe that yes, outputs.logits[0] contains the logits for the non-toxic class. Can you also try using the "bert-based-uncased" tokenizer? I also suggest pasting code using code blocks to help others understand your code!

**Comment by khk-abc** [comment]
using the dataset:
![屏幕截图 2024-01-17 163432](https://github.com/microsoft/TOXIGEN/assets/36502718/816e5137-6d74-4362-a721-a08f9bb0e329)
item['content] means online content,
item['toxic'] ==1 means the content is toxic

using following code:
```
from transformers import AutoTokenizer, AutoModel
import json
from sklearn.metrics import f1_score,precision_score,recall_score,accuracy_score,roc_auc_score
from tqdm import tqdm
import torch
from transformers import pipeline

from transformers import AutoModelForSequenceClassification

toxigen_hatebert = AutoModelForSequenceClassification.from_pretrained("pretrained_models/toxigen_hatebert")
tokenizer = AutoTokenizer.from_pretrained('pretrained_models/bert-base-cased')

# toxigen_hatebert = pipeline("text-classification",
#                             model="pretrained_models/toxigen_hatebert",
#                             tokenizer="pretrained_models/bert-base-cased")
#                             # tokenizer="pretrained_models/toxigen_hatebert")


with open(data_path,'r') as f:
    datas= json.load(f)

# toxigen_hatebert = toxigen_hatebert.to('cuda:3')
# print(list(toxigen_hatebert.parameters())[0].device)

predicts = []
trues = []
contents = []
toxigen_hatebert.eval()
with torch.no_grad():
    for idx,item in enumerate(tqdm(datas)):
        content = item['content']
        # contents.append(content)
        # outputs = toxigen_hatebert(content)
        inputs = tokenizer(content,return_tensors='pt',padding='longest')
        outputs = toxigen_hatebert(**inputs)
        print(outputs)
        trues.append(item['toxic'])
        predicts.append(1-torch.argmax(outputs['logits'][0]).tolist())
        print('true: ', item['toxic'])
        print('pred: ', predicts[-1])


f1=f1_score(y_true=trues,y_pred=predicts)
precision=precision_score(y_true=trues,y_pred=predicts)
recall=recall_score(y_true=trues,y_pred=predicts)
acc=accuracy_score(y_true=trues,y_pred=predicts)
auc=roc_auc_score(y_true=trues,y_score=predicts)

print(f1,precision,recall,acc,auc)
```

**Comment by Thartvigsen** [comment]
Where is the toxic label from?

**Comment by Thartvigsen** [comment]
Yes but can you please point me to where you got the data from and how you processed it. I can't tell what's going on behind the scenes because you're loading these data from disk, not using the huggingface API.

**Comment by Thartvigsen** [comment]
This is perfect, thanks! And can you also share what the outputs are when you compute the metrics above?

**Comment by khk-abc** [comment]
> This is perfect, thanks! And can you also share what the outputs are when you compute the metrics above?
Oops, that is  the outputs with bert-bease-uncased tokenizer:
![image](https://github.com/microsoft/TOXIGEN/assets/36502718/d5a77cef-96c9-4ae3-958a-e36a98230e71)
 that is  the outputs with bert-bease-cased tokenizer:
![image](https://github.com/microsoft/TOXIGEN/assets/36502718/29fd6df2-5589-4287-b3ed-de41b164265e)
that's the key!

**Comment by Thartvigsen** [comment]
Ok perfect, sorry about that! I updated the example in the README
